eng/build.sh

```diff
@@ -26,6 +26,7 @@ usage()
   echo "Test actions:"
   echo "  --testCoreClr              Run unit tests on .NET Core (short: --test, -t)"
   echo "  --testMono                 Run unit tests on Mono"
+  echo "  --testCompilerOnly         Run only the compiler unit tests"
```

Comment:
This is a duplicate of the same flag that we already have for `build.ps1`. Separated into a separate commit, and I will be doing a rebase merge to keep it separate.